js/spProcessMutations.js

```diff
@@ -0,0 +1,119 @@
+// @ts-check
+
+/**
+ * @param {string} accountID
+ * @param {string} clientID
+ * @param {Mutation[]} mutation
+ */
+function spProcessMutations(accountID, clientID, mutation) {
+  exec(() => processMutations(accountID, clientID, mutation));
+}
+
+class PermanentError extends Error {}
+
+/** @typedef {{id: number; name: string; args: any}} Mutation */
+
+// TODO(arv): Should be error and not message
+/** @typedef {{id: number; message: string}} MutationInfo */
+
+/**
+ * @template Value
+ * @typedef {{kind: "Ok", value: Value}} Ok
+ */
+
+/** @typedef {{kind: "BadRequest", message: string}} BadRequest */
+
+/**
+ * @param {string} accountID
+ * @param {string} clientID
+ * @param {Mutation[]} mutations
+ * @return {Promise<BadRequest | Ok<MutationInfo[]>>}
+ */
+async function processMutations(accountID, clientID, mutations) {
+  /** @type {MutationInfo[]} */
+  const infos = [];
+  for (const mutation of mutations) {
```

Comment:
I think that this creates a longer-lived transaction that is strictly necessary to implement Replicache.

We only need the transaction to span the contents of each iteration of this loop, not the entire loop.

I think this is non-ideal for a sample since people will end up copying it. Can we pull the loop back out to csharp? Sorry.

Reply:
of course. I misunderstood.